presto-hive/src/main/java/io/prestosql/plugin/hive/s3/S3SecurityMapping.java

```diff
@@ -71,7 +73,11 @@ public S3SecurityMapping(
         checkArgument(accessKey.isPresent() == secretKey.isPresent(), "accessKey and secretKey must be provided together");
         this.credentials = accessKey.map(access -> new BasicAWSCredentials(access, secretKey.get()));
 
-        checkArgument(!this.allowedIamRoles.isEmpty() || iamRole.isPresent() || credentials.isPresent(), "must provide role and/or credentials");
+        this.useClusterDefault = requireNonNull(useClusterDefault, "useClusterDefault is null")
+                .orElse(false);
+        boolean roleOrCredentialsArePresent = !this.allowedIamRoles.isEmpty() || iamRole.isPresent() || credentials.isPresent();
+        checkArgument(this.useClusterDefault || roleOrCredentialsArePresent, "must either allow useClusterDefault role or provide role and/or credentials");
```

Comment:
this checkArgument seems to be covered by the below, and so could be removed

Reply:
You are right. Below condition with message from this one would be grammatically correct in english.